sync/sync.go

```diff
@@ -42,22 +99,59 @@ func Sync(m cluster.Manifests, repoResources map[string]resource.Resource, clus
 		prepareSyncApply(logger, clusterResources, id, res, &sync)
 	}
 
-	return clus.Sync(sync)
+	if err := clus.Sync(sync, resourceLabels, resourcePolicyUpdates); err != nil {
+		return err
+	}
+	if tracks {
+		orphanedResources := make([]string, 0)
+
+		fmt.Printf("[stack-tracking] scanning stack (%s) for orphaned resources\n", stackName)
+		clusterResourceBytes, err := clus.ExportByLabel(fmt.Sprintf("%s%s", kresource.PolicyPrefix, "stack"), stackName)
```

Comment:
Aha, here it is. This is an abstraction leak: you have to know that cluster/kubernetes uses labels for marking stacks, to be able to export the right things. What's really needed here is to export everything* that has been given a stack, rather than things with a specific label.

*Why not things in a _particular_ stack? Because stacks may come and go -- for example, if they were based on top-level directories, and someone deleted a whole top-level directory.

(Side note: the use of Export in here is a misleading convenience -- what we really want is to be able to get all the things in the cluster; it just happens that Export will do that for us. But let's not try to fix that here.)